client.go

```diff
@@ -15,6 +15,11 @@ func NewKinesisClient() *KinesisClient {
 	return &KinesisClient{svc}
 }
 
+// NewConfiguredKinesisClient returns a configured client to interface with Kinesis stream
+func NewConfiguredKinesisClient(svc *kinesis.Kinesis) *KinesisClient {
```

Comment:
can we use functional options on the `NewKinesisClient` function?

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

```go
type Option func(*KinesisClient)

// WithClient overrides the default Kinesis client
func WithClient(svc *kinesis.Kinesis) Option {
	return func(kc *KinesisClient) {
		kc.svc = svc
	}
}

// NewKinesisClient returns client to interface with Kinesis stream
func NewKinesisClient(opts ...Option) *KinesisClient {
	kc := &KinesisClient{}

	for _, opt := range opts {
		opt(kc)
	}

	if kc.svc == nil {
		kc.svc = kinesis.New(session.New(aws.NewConfig()))
	}

	return kc
}
```

that way the `NewKinesisClient` constructor doesn't need to change

The caller would do something like this:

```go
consumer = consumer.NewKinesisClient(consumer.WithClient(kc))